video/out/opengl/shader_cache.c

```diff
@@ -753,16 +765,19 @@ static void gl_sc_generate(struct gl_shader_cache *sc,
         for (int n = 0; n < sc->params.num_vertex_attribs; n++) {
             const struct ra_renderpass_input *e = &sc->params.vertex_attribs[n];
             const char *glsl_type = vao_glsl_type(e);
+            const char *loc = "";
+            if (sc->ra->glsl_vulkan)
+                loc = mp_tprintf(32, "layout(location=%d) ", n);
```

Comment:
Won't work, because `loc` goes out of scope.

Reply:
(Does that mean it's UB?)

Reply:
(Yes. Makes me wish we could actually use alloca(), which would keep the allocation valid for the whole function.)